app/src/main/java/com/simplecity/amp_library/utils/PlaylistUtils.java

```diff
@@ -216,8 +216,13 @@ public static void clearMostPlayed() {
         void onSave(Playlist playlist);
     }
 
-    public static void makePlaylistMenu(SubMenu sub) {
-        DataManager.getInstance().getPlaylistsRelay()
+    public static void makePlaylistMenu(SubMenu sub, Observable<Long> stopEventTrigger, boolean autoUpdate) {
```

Comment:
It looks like this was just an oversight - the `Observable` remains subscribed and listening for emissions after we've finished making our playlist menu. We only need the first emission from the playlist relay.

Using the stopEventTrigger/LifecycleCallbacks seems like overkill here. I would think instead, we should just be calling `playlistListObservable.first(Collections.emptyList())`, which ensures the `Observable` only emits once and then is disposed.

Reply:
Sometime the submenu need to be updated to show new playlist, for instance: in `QueueFragment`, user click submenu -> create playlist, then open menu again, new playlist should be displayed in the menu.

In case `autoUpdate == false`, it goes to `playlistListObservable.take(1)`, equally with 
`playlistListObservable.first(Collections.emptyList())`.

Reply:
The method `makePlaylistMenu()` is there to create the menu, not keep it up-to-date. I think it is an unwanted/accidental side-effect for it to remain subscribed to changes after it has finished creating the menu.

A better solution, IMO, would be to refactor the method, call it `updatePlaylistMenu()` and have it return the `Observable` - then we can keep a reference to the `Observable` and dispose of it in `onPause()`. 

Passing in the Lifecycle Observable still just feels like an over-engineered solution to me. It's harder to understand what's happening. 

I'll do some refactoring of this PR myself and show you what I think is the cleaner solution.